packages/remix-react/components.tsx

```diff
@@ -548,7 +333,8 @@ export function composeEventHandlers<
  * @see https://remix.run/api/remix#meta-links-scripts
  */
 export function Links() {
-  let { matches, routeModules, manifest } = useRemixEntryContext();
+  let { manifest, routeModules } = useRemixContext();
+  let { matches } = useDataRouterStateContext();
```

Comment:
Most changes to existing/preserved hooks are just grabbing stuff from the new contexts